File: app/src/main/java/org/easyeng/easyeng/MainActivity.java
package org.easyeng.easyeng;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.SparseArray;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public class MainActivity extends AppCompatActivity {

    private SparseArray<Fragment> fragmentMap;
    private SharedPreferences mPrefs;
    private String CURRENT_FRAGMENT = "current_fragment";
    private int currentFragment = R.layout.fragment_home;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        fragmentMap = new SparseArray<>(3);
        fragmentMap.put(R.layout.fragment_home, new HomeFragment());
        fragmentMap.put(R.layout.fragment_puzzles, new PuzzlesFragment());
        fragmentMap.put(R.layout.fragment_words, new WordsFragment());

        final BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation_menu);

        //Get current Screen
        mPrefs = getSharedPreferences("details", 0);
        currentFragment = mPrefs.getInt(CURRENT_FRAGMENT, R.layout.fragment_home);
        //Set selected item in the bottom navigation menu
        int selectedItemId = R.id.action_home;
        if (currentFragment == R.layout.fragment_puzzles)
            selectedItemId = R.id.action_puzzles;
        else if (currentFragment == R.layout.fragment_words)
            selectedItemId = R.id.action_words;

        bottomNavigationView.setSelectedItemId(selectedItemId);
        getSupportFragmentManager().beginTransaction()
                .add(R.id.fragments_container, fragmentMap.get(currentFragment))
                .commit();

        bottomNavigationView.setOnNavigationItemSelectedListener(item -> {
            final int menu_item_id = item.getItemId();
            Fragment fragment = null;
            switch (menu_item_id) {
                case R.id.action_home:
                    fragment = fragmentMap.get(R.layout.fragment_home);
                    currentFragment = R.layout.fragment_home;
                    break;
                case R.id.action_puzzles:
                    fragment = fragmentMap.get(R.layout.fragment_puzzles);
                    currentFragment = R.layout.fragment_puzzles;
                    break;
                case R.id.action_words:
                    fragment = fragmentMap.get(R.layout.fragment_words);
                    currentFragment = R.layout.fragment_words;
                    break;
            }

            return loadFragment(fragment);
        });
    }

    private boolean loadFragment(Fragment fragment) {
        if (fragment == null) return false;
        getSupportFragmentManager()
                .beginTransaction()
                .setCustomAnimations(R.animator.appearing, R.animator.hiding, R.animator.appearing, R.animator.hiding)
                .replace(R.id.fragments_container, fragment, fragment.getTag())
                .commit();
        return true;
    }


    @Override
    protected void onPause() {
        super.onPause();
        SharedPreferences.Editor editor = mPrefs.edit();
        editor.putInt(CURRENT_FRAGMENT, currentFragment);
        editor.apply();
    }
}

File: GettingThingsDone/app/src/main/java/com/example/gettingthingsdone/MainActivity.kt
package com.example.gettingthingsdone

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.gettingthingsdone.db.entity.File


class MainActivity : AppCompatActivity() {

    private var app: GTDApplication? = null

    private lateinit var mainFragment: MainFragment

    private var noteFragment: NoteFragment? = null

    lateinit var sqlEngine: SQLEngine

    var rootId: Long = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        app = application as GTDApplication
        sqlEngine = SQLEngine(app!!)
        createMainFragment()
    }

    private fun createMainFragment() {
        mainFragment = MainFragment()
        supportFragmentManager.beginTransaction()
            .add(R.id.container, mainFragment)
            .commit()
    }

    fun createNoteFragment(file: File?) {
        noteFragment = NoteFragment()
        noteFragment!!.setNote(file)
        supportFragmentManager.beginTransaction()
            .replace(R.id.container, noteFragment!!)
            .addToBackStack("note_fragment")
            .commit()
    }

    override fun onBackPressed() {
        if (noteFragment == null) {
            if (!mainFragment.onBackPressed())
                super.onBackPressed()
        } else {
            if (!noteFragment!!.onBackPressed())
                if (!mainFragment.onBackPressed())
                    super.onBackPressed()
        }
    }
}